build_teamcity

```diff
@@ -2,58 +2,80 @@
 set -x
 set -o errexit -o pipefail
 
-# Fail quickly if docker daemon is not up
-systemctl status docker
+function now { date +"%Y-%m-%dT%H:%M:%S.000" | tr -d '\n' ;}
+function _scope_opened() {
+    set +x
+    echo "##teamcity[blockOpened timestamp='$(now)' name='$1']"
+    set -x
+}
+function _scope_closed() {
+    set +x
+    echo "##teamcity[blockClosed timestamp='$(now)' name='$1']"
+    set -x
+}
 
-# cleanup from previous builds
-# *active.json and *.bootstrap.tar.xz must be cleaned up, otherwise
-# Teamcity starts picking up artifacts from previous builds.
-#
-# We manually clean rather than having TeamCity always clean so that
-# builds are quicker.
-rm -rf dcos-release.config.yaml
-rm -rf artifacts/
-rm -f packages/*.active.json
-rm -f packages/bootstrap.latest
-rm -f packages/*.bootstrap.tar.xz
-rm -f CHANNEL_NAME
-rm -rf build/env
-rm -f dcos_generate_config*.sh
-rm -rf wheelhouse/
+function main() {(
```

Comment:
I don't see what the value of turning this into a function main() is? Seems like it mainly just indents the code more and adds a call to it. No code reuse or the like gained.